src/Lab/Experiments/VulkanTriangle/HelloTriangleApplication.cs

```diff
@@ -257,13 +270,40 @@ private unsafe void CleanupSwapchain()
             _vkSwapchain.DestroySwapchain(_device, _swapchain, null);
         }
 
+        private unsafe string[] GetOptimalValidationLayers()
+        {
+            var layerCount = 0u;
+            _vk.EnumerateInstanceLayerProperties(&layerCount, (LayerProperties*)0);
+
+            var availableLayers = new LayerProperties[layerCount];
+            fixed (LayerProperties* availableLayersPtr = availableLayers)
+            {
+                _vk.EnumerateInstanceLayerProperties(&layerCount, availableLayersPtr);
+            }
+
+            var availableLayerNames = availableLayers.Select(availableLayer => Marshal.PtrToStringAnsi((nint)availableLayer.LayerName)).ToArray();
+            foreach (var validationLayerNameSet in _validationLayerNamesPriorityList)
+            {
+                if (validationLayerNameSet.All(validationLayerName => availableLayerNames.Contains(validationLayerName)))
+                {
+                    return validationLayerNameSet;
+                }
+            }
+
+            return null;
+        }
+
         private unsafe void CreateInstance()
         {
             _vk = Vk.GetApi();
 
-            if (EnableValidationLayers && !CheckValidationLayerSupport())
+            if (EnableValidationLayers)
             {
-                throw new NotSupportedException("Validation layers requested, but not available!");
+                _validationLayers = GetOptimalValidationLayers();
+                if (_validationLayers == null)
```

Comment:
nit: `is null` style

```suggestion
                if (_validationLayers is null)
```